### DISCUSSION TITLE
关于响应文件夹大小查询的速度的猜想

### DISCUSSION THREAD

**Discussion by BrandonStudio** [spencerwooo/onedrive-vercel-index#379] [Ideas]
我不清楚 Onedrive 的文件夹大小是存储还是计算得出，但 Windows 中文件夹大小是每次查询时计算的。如果 Onedrive 的文件夹大小也是查询时计算的话，当文件夹包含大量文件时，查询的响应速度是否会很慢？这样的话子文件夹大小的查询是否会拖慢当前文件夹加载/列出的速度？
这只是猜想。
如果是这样的话是否考虑先列出文件（夹）名和其它属性，再查询每一项的大小？

**Comment by myl7** [comment]
几千几万的量级的话应该完全不需要担心。作为参考：

```shell
$ find .cache -type f | wc -l
<...>
332081
# 十万级的文件量
$ time du .cache -h --max-depth=0
<...>
<...> 0.09s user 0.98s system <...>
# 耗时大概 1s
```

**Comment by spencerwooo** [comment]
> 我不清楚 OneDrive 的文件夹大小是存储还是计算得出

是 API 的 DriveItem 包含的一项属性，应该是存储的而不是实时计算的。（类似于 [Array.prototype.length](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length) 的思想吧。）

https://docs.microsoft.com/en-us/onedrive/developer/rest-api/resources/driveitem
